drv/lpc55-swd/src/main.rs

```diff
@@ -1339,26 +1351,48 @@ impl ServerImpl {
         // write DHCSR to RUN (MATIC + C_DEBUGEN)
         // poll for S_HALT or timeout
 
+        // Search st.com for document "PM0253". Section 2.4.4 describes
+        // the STM32H753 vector table.
+        //
+        // The `endoscope` image has the vector table at offset 0. When loaded
+        // into the SP, that table will be at `endoscope::LOAD` and the SP's
+        // `VTOR` register will be set to that address prior to the SP being
+        // released from reset. Addresses in the vector table are absolute
+        // runtime values.
+        //
+        // The first u32 in the `endoscope` image is the initial stack pointer
+        // value. The second u32 is the initial program counter, a.k.a. the
+        // reset vector.
+
+        const_assert!(ENDOSCOPE_BYTES.len() > 2 * 1024);
+
         // Set SP's Program Counter
-        let sp_reset_vector =
-            u32::from_le_bytes(ENDOSCOPE_BYTES[4..=7].try_into().unwrap_lite());
-        if self
-            .do_write_core_register(Reg::Dr.into(), sp_reset_vector | 1)
-            .is_err()
+        if let Some(sp_reset_vector) = slice_to_le_u32(&ENDOSCOPE_BYTES[4..=7])
```

Comment:
Why did this change?  I found the previous version to be more clear:
```rust
let sp_reset_vector =
    u32::from_le_bytes(ENDOSCOPE_BYTES[4..=7].try_into().unwrap_lite());
```
versus the new code, which hides the `unreachable!` in the `else` of a conditional which is always taken

Reply:
I wanted to get rid of calls to unwrap_lite().

How do you feel about this:

```rust
        // Set SP's Stack Pointer
        if slice_to_le_u32(&ENDOSCOPE_BYTES[0..=3])
            .and_then(|initial_sp| {
                self.do_write_core_register(Reg::Sp, initial_sp).ok()
            })
            .is_some()
        {
            ringbuf_entry!(Trace::WroteSpRegisterFail);
            return Err(());
        }
```

Reply:
Vibes: I prefer the explicit `unwrap_lite`: we should never fail to convert a 4-byte slice into a `u32` and I think it's fine to be explicit about it.  It will be compiled down to nothing ([Godbolt example](https://rust.godbolt.org/z/E4feeq4qr)), so there's no runtime cost.

Reply:
Ok. I'm fine with that.